### PR TITLE
Mark Emails

### PR DESCRIPTION
**Pull request by Jonas-Karl**
this PR allows emails to be marked as read and unread. it is categorized under "organize E-Mails"

### CODE REVIEW

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/MarkEmailsCLIPrompt.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  @Override
+  public void start() {
+    super.start();
+    try {
+      var filteredEmails = provideEmailsUseCase.provideEmails(authUseCase.getSessionUserAddress());
+      BaseCLIPrompt baseCLIPrompt = showActionMenuPrompt(filteredEmails);
+      baseCLIPrompt.start();
+    } catch (MailboxSavingException | MailboxLoadingException e) {
+      printWarning("Could not emails");
```
Comment (jens1o) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/MarkEmailsCLIPrompt.java:42]:
```suggestion
      printWarning("Could not load emails");
```

**Review by jens1o — APPROVED** [review]
code diff looks good to me 👍

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Mailbox.java [hunk 2]
```diff
@@ -45,11 +45,9 @@ public void deliverEmail(Email email, boolean isUnread) {
     this.emails.put(email, Set.of(isUnread ? Label.UNREAD : Label.READ));
   }
 
-  public List<Email> getEmailsWithLabel(Label... labels){
-    return emails.entrySet().stream()
-        .filter(entry -> entry.getValue().containsAll(Set.of(labels)))
-        .map(Map.Entry::getKey)
-        .toList();
+  public List<Email> getEmailsWithLabel(Label... labels) {
+    return emails.entrySet().stream().filter(entry -> entry.getValue().containsAll(Set.of(labels)))
```
Comment (seiferla) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Mailbox.java:49]: correct format ?

**Review by seiferla — APPROVED** [review]

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Mailbox.java (hunk 2 above)
Comment (Jonas-Karl) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Mailbox.java]: Yup! @jens1o messed it up again with his settings :)

**Pull request merged**